### PR TITLE
Limits turning off visibility mask on /new to anonymous mode

### PR DESCRIPTION
**Pull request by jaycee**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6853/

**Comment by jaycee** [comment]
QA:

In GIJoe (or via guiproxy) go to /new. You should be sent to /login and see the login component.

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -2140,7 +2140,9 @@ YUI.add('juju-gui', function(Y) {
           if (this.env.get('connected')) {
             this._switchModelToUUID();
           }
-          this.maskVisibility(false);
+          if (this.anonymousMode) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: Could you add a comment here as to why this was necessary.

**Comment by bac** [comment]
# QA Notes

Bootstrapping on AWS with 2.0.2-yakkety-amd64 and doing `upgrade-gui` with a dist built from this code, I see

https://54.160.66.42:17070/gui/12318d47-a165-4c19-8bcc-4ea35f7077bd/new

being redirected to

https://54.160.66.42:17070/gui/12318d47-a165-4c19-8bcc-4ea35f7077bd/login

and the login dialog displayed.

\o/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6857/

**Comment by jaycee** [comment]
Thanks for the hard work, @bac. 

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**